data-listener/src/main/kotlin/com/playkids/onboarding/data/listener/DataListener.kt

```diff
@@ -0,0 +1,51 @@
+package com.playkids.onboarding.data.listener
+
+import com.playkids.onboarding.sqs.SQSHandler
+import com.playkids.onboarding.sqs.SQSListener
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.launch
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest
+import java.util.concurrent.atomic.AtomicBoolean
+
+class DataListener(
+    private val sqs: SqsAsyncClient,
+    private val queueUrl: String,
+    private val maxNumberOfMessages: Int,
+    private val messageHandler: SQSHandler
+): SQSListener {
+
+    private val isRunning = AtomicBoolean(false)
+
+    override fun stop() = isRunning.set(false)
+
+    override fun start(): Job = GlobalScope.launch {
+        isRunning.set(true)
```

Comment:
you could also use `isActive` from kotlin coroutines, but you would have to change behavior a bit, probably